test/utils.cpp

```diff
@@ -199,6 +253,26 @@ TEST_CASE("run_program()", "[utils]")
 	REQUIRE(utils::run_program(argv, "") == "hello world");
 }
 
+TEST_CASE("resolve_tilde() replaces ~ with the path to the $HOME directory", "[utils]")
+{
+	SECTION("prefix-tilde replaced")
+	{
+		TestHelpers::EnvVar envVar("HOME");
+		envVar.set("test");
+		REQUIRE(utils::resolve_tilde("~") == "test");
+		REQUIRE(utils::resolve_tilde("~/") == "test/");
+		REQUIRE(utils::resolve_tilde("~/dir") == "test/dir");
+		REQUIRE(utils::resolve_tilde("/home/~") == "/home/~");
+	}
+
+	SECTION("unset $HOME variable")
+	{
+		TestHelpers::EnvVar envVar("HOME");
+		envVar.set("test");
```

Comment:
Wait, if `HOME` is set to `test`, while is tilde not expanded? How could this test pass on Travis? It fails for me locally. I'll try to dig deeper in a few hours.

Reply:
That is weird. I'm curious if it has to do with the SECTION macro. 
I could see it being worth it to add some negative tests  to avoid this kind of problem. Although if Travis ignores it there isn't much we can do.

Reply:
Sorry, forgot to mention it here; I already fixed the issue in https://github.com/newsboat/newsboat/commit/6e9660f5dcf5a4c7ca5a7c39b10873359f27186d. In short, test failures didn't fail the Travis build.

It's a separate question how I managed to overlook the error. I still need to reflect on that.